frontend/online-shop/src/components/ProductItem.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Typography } from "@material-ui/core";
+import Grid from "@material-ui/core/Grid";
+import { useProductItemStyles } from "../styles/ProductItemStyle";
+import { Link } from "react-router-dom";
+
+interface ProductProps {
```

Comment:
Great use of an interface for the props. An improvement here would be to have only a `product: Product` instead of all its attributes separated.

Reply:
Do you mean another interface called Product, that would be a property of the ProductProps interface?

Reply:
Yes, that would be a model interface for your backend entity.